commands/scale_cluster.go

```diff
@@ -329,7 +336,7 @@ func validateScaleCluster(args scaleClusterArguments, cmdLineArgs []string, maxB
 	}
 
 	// flag conflicts.
-	if args.numWorkersDesired > 0 && (args.workersMax > 0 || args.workersMin > 0) {
+	if args.numWorkersDesired > 0 && (args.workersMax > 0 || args.workersMin > 0) && (int64(args.numWorkersDesired) != args.workersMax || int64(args.numWorkersDesired) != args.workersMin) {
```

Comment:
Flags are conflicting **only** if the `numWorkers` and `workersMin` / `workersMax` is not equal and not zero.

Reply:
Would it improve it to have the conditions in multiple lines? I can't really think of an improvement without changing a lot of the structure (which I would rather not do as some of it is shared with other commands).

Reply:
We want to reduce complexity to make it understandable. I am interested in when this condition holds true. So, which case does it cover? We can have speaking methods like `args.isScalingEmpty`. That would reduce complexity for `(args.workersMax > 0 || args.workersMin > 0)`, which is already a great deal. 

Then we can use https://github.com/giantswarm/ruleengine or write something like this. 

```
isTrue := true && true
isFalse := false && false

if isTrue && isFalse {
    ...
}
```

Reply:
```suggestion
    desiredWorkersExists := (args.numWorkersDesired > 0)
    scalingParameterIsPresent := (args.workersMax > 0 || args.workersMin > 0)
    desiredWorkersDifferFromMaxNumOfWorkers := (int64(args.numWorkersDesired) != args.workersMax)
    desiredWorkersDifferFromMinNumOfWorkers := (int64(args.numWorkersDesired) != args.workersMin)
    desiredWorkersAreNotAtScalingLimits := (desiredWorkersDifferFromMaxNumOfWorkers || desiredWorkersDifferFromMinNumOfWorkers)

    if desiredWorkersExists && scalingParameterIsPresent && desiredWorkersAreNotAtScalingLimits {
```

Possibly something like this?

Reply:
I followed Tuomas suggestion here for the most part as I think it is a decent solution.
I tried to use the parameters in other validations where it made sense.